pandas/computation/pytables.py

```diff
@@ -197,7 +197,10 @@ def stringify(value):
             return TermValue(int(v), v, kind)
         elif meta == u('category'):
             metadata = com._values_from_object(self.metadata)
-            result = metadata.searchsorted(v, side='left')
```

Comment:
I would do this like:

```
result = metadata.searchsorted(v, side='left')
if not result and v not in metadata:
    result = -1
```
then the 'in' is only done if the result is not found (as opposed to always)